unison-cli/src/Unison/CommandLine/OutputMessages.hs

```diff
@@ -2301,21 +2306,26 @@ showDiffNamespace sn ppe oldPath newPath OBD.BranchDiffOutput {..} =
         (P.syntaxToColor . DeclPrinter.prettyDeclOrBuiltinHeader (HQ'.toHQ hq))
     phq' :: _ -> Pretty = P.syntaxToColor . prettyHashQualified'
     phq :: _ -> Pretty = P.syntaxToColor . prettyHashQualified
-    --
+
     -- DeclPrinter.prettyDeclHeader : HQ -> Either
-    numPatch :: Path.Absolute -> Name -> Numbered Pretty
+    numPatch :: Input.AbsBranchId -> Name -> Numbered Pretty
     numPatch prefix name =
-      addNumberedArg . Name.toString . Name.makeAbsolute $ Path.prefixName prefix name
+      addNumberedArg . Name.toString $ prefixBranchId prefix name
 
-    numHQ :: Path.Absolute -> HQ.HashQualified Name -> Referent -> Numbered Pretty
+    numHQ :: Input.AbsBranchId -> HQ.HashQualified Name -> Referent -> Numbered Pretty
     numHQ prefix hq r = addNumberedArg (HQ.toString hq')
       where
-        hq' = HQ.requalify (fmap (Name.makeAbsolute . Path.prefixName prefix) hq) r
+        hq' = HQ.requalify (prefixBranchId prefix <$> hq) r
 
-    numHQ' :: Path.Absolute -> HQ'.HashQualified Name -> Referent -> Numbered Pretty
+    numHQ' :: Input.AbsBranchId -> HQ'.HashQualified Name -> Referent -> Numbered Pretty
     numHQ' prefix hq r = addNumberedArg (HQ'.toString hq')
       where
-        hq' = HQ'.requalify (fmap (Name.makeAbsolute . Path.prefixName prefix) hq) r
+        hq' = HQ'.requalify (prefixBranchId prefix <$> hq) r
+
+    prefixBranchId :: Input.AbsBranchId -> Name -> Name
+    prefixBranchId branchId name = case branchId of
+      Left sbh -> Name.makeAbsolute $ ShortBranchHash.prefixName sbh name
+      Right pathPrefix -> Name.makeAbsolute . Path.prefixName pathPrefix $ name
```

Comment:
I don't get the Left side of this operation.  Can you give a screenshot of how it shows up to the user?
In my mind, the `Name` it would produce is not valid.  Should it be `Input.AbsBranchId -> Name -> Pretty`?

Reply:
TBH I don't the branch Id path is actually ever used in this scenario (yet) I just needed to add it to satisfy the future possibility that it is.

It should look like this: `#6n740pnnhi.base.List` for example.